### PR TITLE
Add support for TLS 1.3 decryption

### PR DESCRIPTION
**Pull request by adoebber**
Add support for TLS 1.3 disclosure messages. A UDP secret disclosure message can contain one secret or multiple secrets separated by a line break.

Fixes #2

### CODE REVIEW

**Comment by adoebber** [comment]
@Mike-M0 I have implemented the TLS 1.3 decryption in a more general way. Can you check if this works for you?

File: Wireshark/plugins/v2gtlssecret.lua [hunk 1]
```diff
@@ -53,33 +46,40 @@ local function check_version(required_version)
...
-    else
-        return 0
+    local str = buf:raw()
+    local tls_secret_list = {}
+    local info_strings = {}
+    
+    -- one UDP packet may contain several lines, check each line
+    for line in str:gmatch'[^\r\n]+' do
+        -- check if this is really a secret
+        match = line:match'^([%u_]+)%d* %x+ %x+$'
```
Comment (TGruett) [inline Wireshark/plugins/v2gtlssecret.lua:58]: 'local'

File: Wireshark/plugins/v2gtlssecret.lua [hunk 2]
```diff
@@ -107,29 +107,38 @@ function p_v2gtlssecret.dissector(buf,pinfo,root)
...
-                if tls_secret == tls_secret_of_file then
-                    secret_exists_in_file = true
+                local idx_to_delete = {}
+                for idx, tls_secret in ipairs(tls_secret_list) do
+                    if tls_secret == tls_secret_of_file then
+                        table.insert(idx_to_delete, idx)
+                    end
+                end
+                for _, idx in ipairs(idx_to_delete) do
+                    table.remove(tls_secret_list, idx)
```
Comment (TGruett) [inline Wireshark/plugins/v2gtlssecret.lua:126]: you can not use idx here

**Review by TGruett — CHANGES_REQUESTED** [review]

**Comment by Mike-M0** [comment]
> @Mike-M0 I have implemented the TLS 1.3 decryption in a more general way. Can you check if this works for you?

unfortunately the regex seems not to match the data.
Our UDP message contains e.g. the following raw data:
`434c49454e545f48414e445348414b455f545241464649435f53454352455420043704b90404b904b904cb37b90404b90404113804b90420e737b904b9040404040404b904d9370437af37a13738b904b904f537b904b904040a5345525645525f48414e445348414b455f545241464649435f53454352455420043704b90404b904b904cb37b90404b90404113804b90420043b38b9043793370404b904b904b904b904b90404b904af3704b904d937d937b90437b904040a434c49454e545f545241464649435f5345435245545f3020043704b90404b904b904cb37b90404b90404113804b90420b90404040404040437b904b904e737043804040404bd37b90404b9040a5345525645525f545241464649435f5345435245545f3020043704b90404b904b904cb37b90404b90404113804b904200404b904372d38380404510404bd37b904cb37bd37b904b904b904b904f004b90404b904b90437040a`

**Review by TGruett — APPROVED** [review]

**Comment by Mike-M0** [comment]
> @Mike-M0 I have implemented the TLS 1.3 decryption in a more general way. Can you check if this works for you?

works for us. Thanks!

**Comment by adoebber** [comment]
@Mike-M0 When I decode your hex string I get the following:
```
$  echo "43...0a" | xxd -r -p
CLIENT_HANDSHAKE_TRAFFIC_SECRET 7����7��8� �7����77�7�78���7��
SERVER_HANDSHAKE_TRAFFIC_SECRET 7����7��8� ;8�7�7�������7��7�7�7�
CLIENT_TRAFFIC_SECRET_0 7����7��8� �7���78�7��
SERVER_TRAFFIC_SECRET_0 7����7��8� �7-88Q�7��7�7��������7
```
I have implemented the dissector in a way that I can copy the string directly into the key log file. For this the string must look like this:
```
CLIENT_HANDSHAKE_TRAFFIC_SECRET 3b...41 c5...d8
SERVER_HANDSHAKE_TRAFFIC_SECRET ...
```
Does it work with your PR #2? In my opinion, this should not be the case, unless I am missing something.

**Pull request merged**